### DISCUSSION TITLE
make helm part of the system packages

### DISCUSSION THREAD

**Discussion by obeleh** [kcl-lang/kcl#1801] [Ideas]
There are a few helm implementations I saw in [this discussion](https://cloud-native.slack.com/archives/C05TC96NWN8/p1734809916154579) it would be nice if helm would have been a built-in package of KCL. In fact, if I would make a feature (wish)list, this would be definitely on top.

```python
helm.template(
    chart="example",
    target_revision="0.1.0",
    repo_url="https://example.com/charts",
)
```

It would bring KCL to a more central place in my tooling to render YAML. Right now I'm calling both KCL and Helm from Taskfiles. But it would be nice to have KCL build everything 

@zong-zhe

**Comment by mikecameronh1** [comment]
I asked a similar question recently. See https://github.com/orgs/kcl-lang/discussions/1774

My understanding is that you must create a custom [KCL plugin](https://www.kcl-lang.io/docs/reference/plugin/overview) using either Python, Go, Java, or Rust. This plugin would expose the functionality to template a Helm chart with your own values, etc. Say you created a Go plugin with compiled binary `kclx`, you would simply replace all CLI calls from `kcl` to `kclx`.

A great example of this can be found in [cakehappens/kcfoil](https://github.com/cakehappens/kcfoil/tree/main).

Downside of this approach in my opinion is that you will have to maintain a wrapper around the `kcl` CLI.
